### PR TITLE
add checkbox in enhancer options  to use default or custom parameters

### PR DESCRIPTION
**Pull request by cholletNovius**

### CODE REVIEW

File: classes/controller/front.ctrl.php [hunk 1]
```diff
@@ -40,7 +40,12 @@ public function action_show($args = array()) {
         if ($align = \Arr::get($args, 'media_align')) {
             \Arr::set($params, 'attributes.height', $height);
         }
-
+
```
Comment (jguyomard) [inline classes/controller/front.ctrl.php]: trailing space :)

File: classes/driver/oembed.php [hunk 2]
```diff
@@ -52,7 +52,8 @@ public function display($params = array()) {
 		// Gets the player's embedded code
 		$metadatas = (array) $this->metadatas();
 		$display = \Arr::get($metadatas, 'html', '');
-		if (!empty($display)) {
+		//dd($params);
```
Comment (jguyomard) [inline classes/driver/oembed.php]: Hum

File: lang/fr/common.lang.php [hunk 3]
```diff
@@ -42,7 +42,10 @@
     'Right'                                         => 'Droite',
     'Left (floating)'                               => 'Gauche (flottant)',
     'Right (floating)'                              => 'Droite (flottant)',
-
+    'Use default html display'                      => 'Utiliser l\'affichage par défaut',
+    'Explaination checkbox default'                 => 'Si vous décochez cette case, les paramètres ci-dessous seront appliqués',
+    'Custom parameters only for Youtube and Soundcloud' => 'Paramètres personnalisés (seulement pour Youtube et Soundcloud)',
+
```
Comment (jguyomard) [inline lang/fr/common.lang.php]: trailing space :)

File: classes/controller/front.ctrl.php [hunk 4]
```diff
@@ -40,7 +40,12 @@ public function action_show($args = array()) {
         if ($align = \Arr::get($args, 'media_align')) {
             \Arr::set($params, 'attributes.height', $height);
         }
-
+    
+        //use default embed code
+        if ($defaultEmbed = \Arr::get($args, 'media_use_default_embed_display')) {
```
Comment (jguyomard) [inline classes/controller/front.ctrl.php]: No assignation in a `if` please

**Review by jguyomard — CHANGES_REQUESTED** [review]

File: classes/driver/soundcloud.php [hunk 5]
```diff
@@ -20,4 +20,18 @@ class Driver_Soundcloud extends Driver_Oembed {
 	public function compatible() {
         return ($this->host(false) == 'soundcloud.com' && parent::compatible());
 	}
+    
+    /**
```
Comment (Meroje) [inline classes/driver/soundcloud.php]: this file has mixed indents, would be nice to fix that

File: classes/driver/soundcloud.php [hunk 6]
```diff
@@ -20,4 +20,18 @@ class Driver_Soundcloud extends Driver_Oembed {
...
+     * Returns the HTML code to embed the online media
+     *
+     * @param array $params
+     * @return mixed|string
+     */
+    public function display($params = array()) {
+        return parent::display(\Arr::merge(array(
+            'attributes'	=> array(
+                'src' => '//w.soundcloud.com/player/?visual=true&url='.$this->url().'&show_artwork=true',
```
Comment (Meroje) [inline classes/driver/soundcloud.php]: protocol relative URIs are bad, just use https

**Review by Meroje — CHANGES_REQUESTED** [review]

**Comment by cholletNovius** [comment]
up

**Review by jguyomard — APPROVED** [review]

**Comment by cholletNovius** [comment]
Meroje ? please

**Review by Meroje — APPROVED** [review]

**Pull request merged**